### ISSUE TITLE
hangs on restarting redis

### ISSUE THREAD

**Issue by YAmikep** [DavidWittman/ansible-redis#39]
Hi,

Very nice Ansible script. Good job. Thank you, it helps a lot.

However, when I run it, it hangs on the "restart redis" task, I have to stop it manually:

```
NOTIFIED: [rolebooks/db/redis | restart redis] ********************************
```

Do you have the same behavior?

When I look at the services running on the targeted machine, it seems to be ok:

```
vagrant@vagrant-ubuntu-trusty-64:/$ ps aux | grep redis
root      6851  0.0  0.1   4444   648 ?        S    01:10   0:00 /bin/sh /etc/init.d/redis_6379 start
redis     6855  0.3  0.4  36740  2388 ?        Sl   01:10   0:00 /opt/redis/bin/redis-server 127.0.0.1:6379
```

Here is how I used the script:

```
- hosts: all
  sudo: yes  
  roles:
    - { 
        role: rolesbooks/db/redis,
        redis_version: "2.8.17",
        redis_bind: "127.0.0.1",
        redis_port: "6379",
        redis_logfile: "{{project_dirs.logs}}/redis.log",
        redis_databases: "1",
        redis_daemonize: "no",
        redis_maxmemory: "100mb",
        redis_maxmemory_policy: "allkeys-lru",
      }   
```

Thanks.

**Comment by stephansarver** [comment]
I am experiencing the same issue but it happens when using the 
`redis_password: somelongpassword property`

From what I can tell it looks like in the init script that resides in /etc/init.d/redis_6379 is expecting a variable to be set when using password auth.
`
if [ -n "$REDIS_PASSWORD" ]; then
    CLIEXEC="${CLIEXEC} -a ${REDIS_PASSWORD}"
fi
`

so in the stop block of code

`
  stop)
        if [ -f "$PIDFILE" ]; then
            PID=$(cat "$PIDFILE")
            log_daemon_msg "Stopping $NAME..."
            $CLIEXEC shutdown
            while [ -x /proc/${PID} ]; do
                log_daemon_msg "Waiting for Redis to shutdown ..."
                sleep 1
            done
            log_end_msg 0
        else
            log_daemon_msg "$NAME is not running"
            log_end_msg 0
        fi
        ;;
`

it issues the command
`
 /opt/redis/bin/redis-cli -p 6379
`
that fails on authentication and enters an infinite loop.

My question is should I be setting this value somewhere that is not obvious from the documentation?

**Comment by frutality** [comment]
I know it's "a bit" old issue, but I just experienced **exactly** same behaviour as @YAmikep 

The only difference is that I was not daemonizing redis. I actually only added `redis_log` variable, that's all. Steps, exactly what I did:

1. Run playbook with this role (default settings, latest role version). At this moment, I did not check if redis is working.
2. Later, I realized that my app cannot connect redis. Error was "_MISCONF Redis is configured to save RDB snapshots, but is currently not able to persist on disk_"
3. I decided to check redis logs and realized that there is no logfile by default. So I added `redis_logfile: '/var/log/redis.log'` and ran playbook again.
4. It hangs at `restart redis`. ^C, run playbook again - now it hangs at `ensure redis is running`.
5. Tried to run few more times with same result.
6. Removed `redis_logfile` and ran it again. Still hangs.
7. Connected to server via ssh and killed `redis-server` process. It restarted and.. my app now can connect redis.
8. Just to make sure, ran playbook once again - it worked just fine, no hangs.

OS is Ubuntu 16.04 (server), Xubuntu 16.04 (client), ansible is 2.4.2.0, python is 2.7.12.

Clearly don't know where was the problem. Posting this to help anyone who finds himself in same situation.
